lib/pages/filter/view/filter_page.dart

```diff
@@ -175,31 +140,61 @@ class FilterPageState extends State<FilterPage> {
           },
         )
       ],
-      body: filter == null
-          ? Center(child: CircularProgressIndicator())
-          : ListView(
-              children: <Widget>[
-                    if (widget.info != null)
-                      Padding(
-                        padding: const EdgeInsets.only(
-                            left: 10.0, right: 10.0, top: 10.0),
-                        child: IconText(
-                          icon: Icons.info,
-                          text: widget.info,
-                          style: Theme.of(context).textTheme.bodyText1,
-                        ),
-                      ),
-                    if (widget.hint != null)
-                      Padding(
-                        padding: const EdgeInsets.only(
-                            left: 10.0, right: 10.0, top: 5.0),
-                        child: Text(
-                          widget.hint,
-                          style: TextStyle(color: Theme.of(context).hintColor),
-                        ),
-                      )
-                  ] +
-                  widgets),
+      body: FutureBuilder<Filter>(
+          future: Provider.of<FilterProvider>(context).fetchFilter(context),
+          builder: (context, snapshot) {
+            if (snapshot.hasData) {
+              filter ??= snapshot.data;
+              List<Widget> widgets = [SizedBox(height: 10.0)];
+
+              Map<int, List<Widget>> optionsByLevel = {};
+              _buildTree(node: filter.root, optionsByLevel: optionsByLevel);
+              for (var i = 0; i < filter.localizedLevelNames.length; i++) {
+                if (optionsByLevel[i] == null || optionsByLevel.isEmpty) {
+                  break;
+                }
```

Comment:
maybe use For(var something in  filter.localized)

Reply:
I need the index for `optionsByLevel`.